pupil-api/src/controllers/auth.controller.ts

```diff
@@ -13,15 +32,21 @@ class AuthController {
     }
 
     const { pupilPin, schoolPin } = req.body
-
+    let data
     try {
-      const data = await pupilAuthenticationService.authenticate(pupilPin, schoolPin)
+      if (this.useRedisAuth) {
+        data = await this.redisAuthService.authenticate(schoolPin, pupilPin)
+        if (data === null) {
+          return apiResponse.unauthorised(res)
+        }
+      } else {
+        data = await pupilAuthenticationService.authenticate(pupilPin, schoolPin)
+      }
+      console.log(typeof data)
```

Comment:
debug?

Reply:
added in the throes of despair, and forgotten in the moment of joy when the fix was realised 😄